server/src/main/java/org/opensearch/tasks/consumer/TopNSearchTasksLogger.java

```diff
@@ -36,17 +36,20 @@ public class TopNSearchTasksLogger implements Consumer<Task> {
     private static final Logger SEARCH_TASK_DETAILS_LOGGER = LogManager.getLogger(TASK_DETAILS_LOG_PREFIX + ".search");
 
     // number of memory expensive search tasks that are logged
-    private static final Setting<Integer> LOG_TOP_QUERIES_SIZE_SETTING = Setting.intSetting(
+    public static final Setting<Integer> LOG_TOP_QUERIES_SIZE_SETTING = Setting.intSetting(
         LOG_TOP_QUERIES_SIZE,
         10,
+        1,
+        25,
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
 
     // frequency in which memory expensive search tasks are logged
-    private static final Setting<TimeValue> LOG_TOP_QUERIES_FREQUENCY_SETTING = Setting.timeSetting(
+    public static final Setting<TimeValue> LOG_TOP_QUERIES_FREQUENCY_SETTING = Setting.timeSetting(
```

Comment:
We also need to add settings update consumer to make the settings dynamic, and for that we need object of ClusterSettings here.